libjolie/src/jolie/lang/parse/ast/types/TypeChoiceDefinition.java

```diff
@@ -0,0 +1,96 @@
+package jolie.lang.parse.ast.types;
+
+import jolie.lang.NativeType;
+import jolie.lang.parse.OLVisitor;
+import jolie.lang.parse.context.ParsingContext;
+import jolie.util.Range;
+
+import java.util.Map;
+import java.util.Set;
+
+public class TypeChoiceDefinition extends TypeDefinition {
+    private final TypeDefinition T1;
+    private TypeDefinition T2;
+    //private Map< String, TypeDefinition > subTypes = null;
+    //private boolean untypedSubTypes = false;
+    private Range t2Cardinality;
+
+    public TypeChoiceDefinition(ParsingContext context, String id, Range cardinality, TypeDefinition T1)
+    {
+        super(context, id, cardinality );
+        this.T1 = T1;
+    }
+
+    public TypeChoiceDefinition setT2(TypeDefinition T2, Range cardinality) {
+        this.T2 = T2;
+        this.t2Cardinality = cardinality;
+        return this;
+    }
+
+    public Range getT2Cardinality(){
```

Comment:
Why is this needed? Can't we just call getT2.cardinality() ?

The same goes for all other methods accessing T1 and T2.